### PR TITLE
COSI-58-add-dependabot-and-codecov-in-ci

### PR DESCRIPTION
**Pull request by anurag4DSB**
Implementing Dependabot for automatic dependency updates and integrating Codecov to monitor and maintain code coverage percentages.: https://app.codecov.io/github/scality/cosi-driver/

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    labels:
+      - "gomod"
+      - "dependencies"
+
+
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+      time: "08:00"
```
Comment (anurag4DSB) [inline .github/dependabot.yml:30]: Its intentional not to put timezone here

**Review by dvasilas — APPROVED** [review]

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    reviewers:
+      - "scality/object"
```
Comment (fredmnl) [inline .github/dependabot.yml:32]: What does this do exactly?

**Review by fredmnl — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (fredmnl) [reply .github/dependabot.yml:32]: Does it choose random reviewers in this group? Or just assign the group whatever that means?

Comment (anurag4DSB) [reply .github/dependabot.yml:32]: I think reviewers should include scality/object squad team folks to merge. (mandatory)

Comment (anurag4DSB) [reply .github/dependabot.yml:32]: Trying it out

**Pull request merged**